test/spsc_queue_test.cpp

```diff
@@ -405,3 +406,161 @@ BOOST_AUTO_TEST_CASE( spsc_queue_reset_test )
 
     BOOST_REQUIRE(f.empty());
 }
+
```

Comment:
Shouldn't there also be a test for a move_only_type like shared_ptr?

Reply:
+1